src/main/java/HttpServerThread.java

```diff
@@ -0,0 +1,48 @@
+package com.td.HttpServer;
+
+public class HttpServerThread implements Runnable {
```

Comment:
This works, _but_, your entire `HttpServerThread` class is now tied to the java threading library. That's typically fine (actually you _have_ to implement `Runnable` to use threads) but try to look for better ways to corner this dependency.

`HttpServerThread` already has one responsibility (coordinating the retrieval of a request, fetching a corresponding handler for it, and then piping the response back), and mixing this in does not feel right.

Is it possible to separate all of this "coordination" code from the actual "running this thing somewhere (in a thread)"?